plugins/woocommerce/src/Enums/OrderStatus.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace Automattic\WooCommerce\Enums;
+
+/**
+ * Enum class for all the order statuses.
+ *
+ * For a full documentation on the public order statuses, please refer to the following link:
+ * https://woocommerce.com/document/managing-orders/order-statuses/
+ */
+final class OrderStatus {
```

Comment:
Hey @wjrosa this question came up in the similar PR to Subscriptions, but do we need to also create constants for each of the `wc-*` versions of order statuses as well i.e.

```
const WC_PENDING = 'wc-pending';
```

Apologies if this has already been discussed and I've missed it :)

Reply:
Given that `wc-' prefix handling occurs in several places across the core, we are good with this set of constants as the main source of truth.

Reply:
If it makes sense, I can draft the following PR only with the legacy statuses (on a separate class).